### ISSUE TITLE
[panic]: index out of bounds: the len is 1 but the index is 2

### ISSUE THREAD

**Issue by qarmin** [charliermarsh/ruff#3425]
0.0.254
Config with all rules

```
thread 'main' panicked at 'index out of bounds: the len is 1 but the index is 2', crates/ruff/src/source_code/locator.rs:71:9
```

[_bgcolor (8th copy)15.py.zip](https://github.com/charliermarsh/ruff/files/10936022/_bgcolor.8th.copy.15.py.zip)

**Comment by qarmin** [comment]
I have more crashes in other lines in locator.rs, but probably this are duplicates of this issue

**Comment by charliermarsh** [comment]
Def agree that we shouldn't be panicking, but is that valid Python code? How would you describe the contents of that file? I'm trying to make sure I'm not misinterpreting the issue.

![Screen Shot 2023-03-09 at 4 14 38 PM](https://user-images.githubusercontent.com/1309177/224161327-6d648b80-7657-4314-a5ec-0a45b8b7c0fb.png)

**Comment by qarmin** [comment]
This is found by fuzzer.
I provided valid python file and changed some bytes/charaters to find possible crashes

**Comment by qarmin** [comment]
Smaller reproducer - [a.py.zip](https://github.com/charliermarsh/ruff/files/10947795/a.py.zip)

**Comment by charliermarsh** [comment]
Okay, this is now resolved via #3454 and #3439.
